extensions/vscode/src/utils/uri.ts

```diff
@@ -0,0 +1,16 @@
+import * as vscode from 'vscode';
+import { dirname } from 'path';
+
+export function uriDirname(uri: vscode.Uri): vscode.Uri {
+  if (uri.path.length === 0 || uri.path === '/') {
+    return uri;
+  }
+
+  let path = dirname(uri.path);
+
+  if (path.length === 1 && path.charCodeAt(0) === 46 /* CharCode.Period */) {
```

Comment:
Why not just `if (path === ".")` ?

Reply:
Oh good point! I copied that from https://www.npmjs.com/package/vscode-uri to test it but forgot to remove it. I'll add a dependency to `vscode-uri` and use the function from that package directly instead.